llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -470,6 +470,61 @@ def IREELinalgExt_ReverseOp : IREELinalgExt_Op<"reverse", [
   }];
 }
 
+def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
+  DeclareOpInterfaceMethods<LinalgExtInterface>
+]> {
+  let summary = "Top-K operator";
+  let description = [{
+   A Top-K operation for N-D tensors.
+
+   Accepts two N-D tensor inputs consisting of values (f32) and indices of those
```

Comment:
Why does this have to be f32. Could be any value.

Reply:
Actually it can be any value. The code only checks that the input/output types match. Updating this comment.